File: version_java/src/main/java/com/example/version_java/room/dao/CriminalDao.java
package com.example.version_java.room.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.version_java.room.entity.CriminalEntity;

import java.util.List;

@Dao
public interface CriminalDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Long registerCriminalEntity(CriminalEntity criminalEntity);

    @Query("SELECT * FROM criminal_table")
    List<CriminalEntity> getAll();

    @Query("SELECT * FROM criminal_table WHERE `name` = (:name) ")
    CriminalEntity getCriminalEntity(String name);
}

File: app/src/main/java/com/example/crimenotification/room/dao/CriminalDao.kt
package com.example.crimenotification.room.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.crimenotification.room.entity.CriminalEntity

@Dao
interface CriminalDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun registerCriminalEntity(criminalList: CriminalEntity): Long

    @Query("SELECT * FROM criminal_table")
    fun getAll(): List<CriminalEntity>

}